### PR TITLE
refactor most validateProps errors to warnings

### PR DESCRIPTION
**Pull request by giladgray**
- Refactor most Popover errors to warnings
- Refactor PopoverExample to avoid warnings
- Refactor most errors into warnings
- Refactor RadioGroup errors to support non-Radio children
- Remove non-error tests, fix others

### CODE REVIEW

File: packages/core/test/controls/radioGroupTests.tsx [hunk 1]
```diff
@@ -59,15 +59,14 @@ describe("RadioGroup", () => {
...
+    it("uses options if given both options and children", () => {
+        const group = mount(<RadioGroup onChange={emptyHandler} options={[]}><Radio value="one" /></RadioGroup>);
+        assert.lengthOf(group.find(Radio), 0);
     });
 
-    it("throws error if given non-Radio children", () => {
-        assert.throws(() => {
-            mount(<RadioGroup onChange={emptyHandler}><address /><Radio /></RadioGroup>);
-        });
+    it("renders non-Radio children too", () => {
```
Comment (adidahiya) [inline packages/core/test/controls/radioGroupTests.tsx]: 👌

File: packages/core/src/components/toast/toaster.tsx [hunk 2]
```diff
@@ -94,7 +94,7 @@ export class Toaster extends AbstractComponent<IToasterProps, IToasterState> imp
      */
     public static create(props?: IToasterProps, container = document.body): IToaster {
         if (props != null && props.inline != null) {
-            console.warn(TOASTER_INLINE_WARNING);
+            console.warn(TOASTER_WARN_INLINE);
```
Comment (adidahiya) [inline packages/core/src/components/toast/toaster.tsx]: need to add an `isNodeEnv` check around this?

File: packages/core/src/components/popover/popover.tsx [hunk 3]
```diff
@@ -560,7 +557,7 @@ export class Popover extends AbstractComponent<IPopoverProps, IPopoverState> {
 
     private updateTether() {
         if (this.state.isOpen && !this.props.inline && this.popoverElement != null) {
-            const { constraints, position, tetherOptions, useSmartPositioning } = this.props;
+            const { constraints, position, tetherOptions = {}, useSmartPositioning } = this.props;
```
Comment (adidahiya) [inline packages/core/src/components/popover/popover.tsx]: props defaults are usually set with `defaultProps`... why are you changing this?

Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: so here's my thinking. 

1. default props are great for special _values_, like `false` or `Position.BOTTOM`, but they're not good for objects. for instance, #973 fixed a subtle bug due to the same `{}` instance being reused _for every instance_. i'm now leaning towards omitting `{}` and `[]` from default props as a rule.

2. there's a warning for using this prop `tetherOptions` with `inline` and it's much easier to detect prop presence if the default value is `undefined`. with the previous `{}` there was no good cheap way to determine if the user had/not specified the prop.

Comment (adidahiya) [inline packages/core/src/components/popover/popover.tsx]: Cool, that makes sense. We should document these conventions somewhere.

Comment (giladgray) [inline packages/core/src/components/popover/popover.tsx]: agreed, but where? any ideas?

**Pull request merged**